core/src/packets/ethernet.rs

```diff
@@ -522,4 +529,22 @@ mod tests {
 
         assert_eq!(EthernetHeader::size_of(), ethernet.len());
     }
+
+    /// Because `EthernetHeader` is a union, writing it directly into the
+    /// buffer causes the 8 bytes following the 14-byte tagless header to be
+    /// corrupted. The following code caused the bug.
+    ///
+    /// ```
+    /// mbuf.extend(offset, 14)?;
+    /// mbuf.write_data(offset, &EthernetHeader::default())?;
+    /// ```
+    #[capsule::test]
+    fn bug_push_ethernet_corrupts_buffer() {
```

Comment:
@drunkirishcoder really want to go w/ this name/comment, and not something more general?

Reply:
as discussed, this test is very specific to this piece of bad code. otherwise, the test wouldn't make sense without this additional context. will mark the test with the release version that has the bug and keep the comments.